subprojects/ide-native/src/main/java/org/gradle/ide/xcode/internal/XcodeTarget.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.ide.xcode.internal;
+
+import org.gradle.api.Named;
+import org.gradle.ide.xcode.internal.xcodeproj.PBXTarget;
+import org.gradle.internal.id.UUIDGenerator;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @see <a href="https://developer.apple.com/library/content/featuredarticles/XcodeConcepts/Concept-Schemes.html">XCode Scheme Concept</a>
+ */
+public class XcodeTarget implements Named {
+    private static final UUIDGenerator ID_GENERATOR = new UUIDGenerator();
```

Comment:
Should inject this, and should probably use a stable id for a given target instead of a random UUID each time the Xcode files are generated. There's a `GidGenerator` class that also seems to be serving the same purpose. We should have one way to generate the IDs of things in the project.